scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.jsx

```diff
@@ -106,9 +105,7 @@ SpellcheckerContextMenuComponent.propTypes = {
  * @description Maps the values from state to the component callback type props.
  */
 const mapDispatchToProps = (dispatch) => ({
-    addWord: (word) => dispatch(actions.ignoreWord(word)),
-    ignoreWord: (word) => dispatch(actions.addWord(word)),
-    replaceWord: (word) => setTimeout(() => dispatch(actions.replaceWord(word)), 0),
+    replaceWord: (word) => setTimeout(() => dispatch(actions.replaceWord(word)), 0)
```

Comment:
Just to follow up, I've tried the above and it doesn't work. There seems to be a problem with synchronising editor states, so the `setTimeout` part seems to be needed.